W-17017575-insights-hyperforce-duke (#314)
mulesoft/docs-monitoring: modules/ROOT/pages/alerts.adoc

= Getting Alerts in Anypoint Monitoring
:page-aliases: alerts-using.adoc, alerts-config.adoc

//feature tier:
include::reuse::partial$billing/pricing.adoc[tag=featureAdvSubReqMonitoring]

Alerts indicate whether a resource (such as a Mule app or API) is behaving as expected or exceeding a defined threshold. 

[[alerts]]
image::intro-alerts.png["Alerts in Anypoint Monitoring"]

> You can create different types of alerts:

Reviewer: ```suggestion
You can create these types of alerts:
```

* xref:basic-alerts.adoc[Basic alerts] for servers, Mule apps, and APIs
* xref:advanced-alerts.adoc[Advanced alerts] for graphs in custom dashboards in Anypoint Monitoring
* <<other-alerts, Other alerts in API Manager or Runtime Manager>>:

** Operational alerts for APIs through API Manager
** Operational alerts for server and app events through Runtime Manager

For alerts in Anypoint Monitoring, you can: 

** Enable or disable advanced alerts configured for graphs in custom dashboards
** Filter by <<alert_status, alert status>>
** Filter by <<alert_severity, alert severity>>

== Alert Thresholds

For basic and advanced alerts in Anypoint Monitoring, thresholds are checked every five minutes. The evaluated query is the average value of the metric in the last five minutes. For advanced alerts, the evaluated metric is the one referenced in the *General* tab for the graph. If the value of the metric passes the threshold, the alert triggers. The alert triggers only if the alert state changes (from `OK` state to `Alerting` state or vice versa). If you change the email address of the alert recipient, the recipient doesn't receive an email until the alert triggers.

== Alert Limits

> The number of active alerts allocated to your organization varies based on your organization's pricing model.

Reviewer: ```suggestion
The number of active alerts allocated to your organization depends on your organization's pricing model.
```

Depending on the value, the alert transitions to `OK` or `Alerting`.
* Disabling an alert causes the alert to transition to `Disabled`.

[[other-alerts]]
== Add API Manager and Runtime Manager Alerts

> In addition to creating basic and advanced alerts in Anypoint Monitoring, you can also navigate from Anypoint Monitoring to alert creation pages in API Manager and Runtime Manager. 

Reviewer: ```suggestion
In addition to creating basic and advanced alerts in Anypoint Monitoring, you can navigate from Anypoint Monitoring to the alert-creation pages in API Manager and Runtime Manager. 
```

image::alerts-menu.png["Alerts menu options to set up basic, advanced, or other types of alerts"]

//TODO: VERIFY NOTE

> NOTE: Alerts created in API Manager and Runtime Manager aren't listed in Anypoint Monitoring *Alerts*.

Reviewer: Avoid using UI elements as parts of speech... 

NOTE: Alerts created in API Manager and Runtime Manager aren't listed in the Anypoint Monitoring *Alerts* page.

or

NOTE: Alerts created in API Manager and Runtime Manager aren't listed in Anypoint Monitoring alerts.

Author: replacing with option 1

> NOTE: Alerts created in API Manager and Runtime Manager aren't listed in Anypoint Monitoring *Alerts*.

Author: ```suggestion
NOTE: Alerts created in API Manager and Runtime Manager aren't listed in the Anypoint Monitoring Alerts page.
```

> . From *Alerts* in Anypoint Monitoring, click *+ New alert*.

Reviewer: I think we don't include punctuation from UI elements.

Author: I'm not sure if that's the case where it's for semantic purposes. The "+" on the button means Add. I think it's necessary to add it so that it provides that meaning conveyed by the button text.

Author: addressed some alt text. for the rest: [W-17361464](https://gus.lightning.force.com/a07EE000026JsovYAC)

@dukesphere

> . Select the *Other* option for API Manager or Runtime Manager to access its alert configuration page. 

Reviewer: ```suggestion
. Select *Other* for API Manager or Runtime Manager to access its alert configuration page. 
```

> * *Create an alert for your API in API Manager*

Reviewer: Are these two bold bullets UI options? (Bold won't be translated.)

Author: They are in the UI that way.

+

> Alerts created in API Manager aren't listed in Anypoint Monitoring *Alerts*.

Reviewer: See Line 81. 

You could create a partial for this.

Author: Yeah, will look into additional partials after release.

TODO: @dukesphere

Author: [W-17361464](https://gus.lightning.force.com/a07EE000026JsovYAC)

Author: Followup: Deleting second mention. Line 81 isn't repeated bc the step is specific to creating an alert in API Manager, not RTM, so it doesn't match the note. The note is prominent and close to the procedure, so include repetition prob not necessary.

* *Create an alert for your API in Runtime Manager*

. To create these alerts, proceed to:

> * xref:runtime-manager::alerts-on-runtime-manager.adoc[Runtime Manager Alerts]

Reviewer: Hard-coded link text OK?

Author: Yes, this is to clarify that this is for Runtime Manager vs. just "Alerts" to distinguish which alerts we're talking about.
